### PR TITLE
chore: Improve FunctionalTestWriter

### PR DESCRIPTION
**Pull request by richard-bogdan-bitrise**
FunctionalTestWriter does not require reset from now on.

APM-2922

### CODE REVIEW

**Comment by richard-bogdan-bitrise** [comment]
So it was bugging me, that we have to reset the writer after the functional tests have run, because it is possible to forget that, so I updated a the code a bit, now it does not requires the reset.

File: trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java [hunk 1]
```diff
@@ -17,7 +18,7 @@
     private static final char[] indentation = new char[]{' ', ' ', ' '};
     private static final char[] escapeChar = new char[]{27};
     private final char[] coloringCode;
-    private final int extraLength;
+    private final char[] coloringEnd = concatEscapeChar("[39;49m".toCharArray());
```
Comment (daniellevass-bitrise) [inline trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java:21]: what does `[39;49m` mean here?

File: trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java [hunk 2]
```diff
@@ -106,7 +91,7 @@ public void write(@NonNull final char[] chars, final int i, final int i1) {
 
     /**
      * Inserts to the start the coloring chars to a given array of chars. Please note that adding the coloring will
-     * stay until changed. To reset it to the default please use the {@link #reset()} method.
+     * stay until changed.
```
Comment (daniellevass-bitrise) [inline trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java:94]: nit: not sure this javadoc makes sense "the coloring will stay until changed" - i think the "until changed" isn't helpful - how do we change it? Perhaps something like "until a new color is requested, or the functional tests complete"?

**Review by daniellevass-bitrise — APPROVED** [review]
Thanks for taking the time to improve this! 

Left two comments about making things clearer to others who read this code :)

File: trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java (hunk 2 above)
Comment (richard-bogdan-bitrise) [reply trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java:94]: Thanks for spotting it, removed them.

File: trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java (hunk 1 above)
Comment (richard-bogdan-bitrise) [reply trace-gradle-plugin/src/functionalTest/java/io/bitrise/trace/plugin/util/FunctionalTestWriter.java:21]: That is the required ANSI code to end the coloring.

**Pull request merged**